docs/optlib.rst

```diff
@@ -181,269 +206,413 @@ Tips about writing option file
 
-Next example is based on an issue #219 posted by @andreicristianpetcu::
+The newly introduced ``--mline-regex-<LANG>`` is similar to ``--regex-<LANG>``
+except the pattern is applied to the whole file's contents, not line by line.
+
+This example is based on an issue #219 posted by @andreicristianpetcu:
+
+.. code-block:: java
+	
+	// in input.java:
 
-	$ cat input.java
 	@Subscribe
 	public void catchEvent(SomeEvent e)
 	{
 	return;
 	}
 
-
 	@Subscribe
 	public void
 	recover(Exception e)
 	{
 	return;
 	}
 
-	$ cat spring.ctags
+The above java code is similar to the Java `Spring <https://spring.io>`_
+framework. The ``@Subscribe`` annotation is a keyword for the framework, and the
+developer would like to have a tag generated for each method annotated with
+``@Subscribe``, using the name of the method followed by a dash followed by the
+type of the argument. For example the developer wants the tag name
+``Event-SomeEvent`` generated for the first method shown above.
+
+To accomplish this, the developer creates a :file:`spring.ctags` file with
+the following:
+
+.. code-block:: perl
+
+	# in spring.ctags:
 	--langdef=javaspring
-	--langmap=javaspring:.java
+	--map-javaspring:+.java
 	--mline-regex-javaspring=/@Subscribe([[:space:]])*([a-z ]+)[[:space:]]*([a-zA-Z]*)\(([a-zA-Z]*)/\3-\4/s,subscription/{mgroup=3}
-	--excmd=mixed
 	--fields=+ln
 
+And now using :file:`spring.ctags` the tag file has this:
+
+.. code-block:: console
+
 	$ ./ctags -o - --options=./spring.ctags input.java
 	Event-SomeEvent	input.java	/^public void catchEvent(SomeEvent e)$/;"	s	line:2	language:javaspring
 	recover-Exception	input.java	/^    recover(Exception e)$/;"	s	line:10	language:javaspring
 
+Multiline pattern flags
+......................................................................
+
+.. note:: These flags also apply to the experimental ``--_mtable-regex-<LANG>``
+	option described later.
+
 ``{mgroup=N}``
 
-	This tells the pattern should be applied to whole file
-	contents, not line by line.  ``N`` is the number of a group in the
-	pattern. The specified group is used to record the line number
-	and the pattern of tag. In the above example 3 is
-	specified. The start position of the group 3 within the whole
-	file contents is used.
+	This flag indicates the pattern should be applied to the whole file
+	contents, not line by line. ``N`` is the number of a capture group in the
+	pattern, which is used to record the line number location of the tag. In the
+	above example ``3`` is specified. The start position of the regex capture
+	group 3, relative to the whole file is used.
+
+.. warning:: You **must** add an ``{mgroup=N}`` flag to the multi-line
+	``--mline-regex-<LANG>`` option, even if the ``N`` is ``0`` (meaning the
+	start position of the whole regex pattern). You do not need to add it for
+	the multi-table ``--_mtable-regex-<LANG>``.
+
+.. Q: isn't the above restriction really a bug? I think it is. I should fix it.
```

Comment:
Yes, it looks a bug.
Can I ask you to open an issue with an example (or `.b` unit test).